src/test/scala/uk/gov/nationalarchives/signcookies/LambdaTest.scala

```diff
@@ -0,0 +1,100 @@
+    val byteArrayCaptor: ArgumentCaptor[Array[Byte]] = ArgumentCaptor.forClass(classOf[Array[Byte]])
+
+    new Lambda().handleRequest(inputStream, outputStream(byteArrayCaptor), null)
+
+    decode[LambdaResponse](byteArrayCaptor.getValue.map(_.toChar).mkString).toOption.get.statusCode should equal(401)
+  }
+
+  "the lambda" should "return 401 if the token is not a valid jwt token" in {
+    val accessToken = "faketoken"
+    val inputStream = getInputStream(accessToken)
+    val byteArrayCaptor: ArgumentCaptor[Array[Byte]] = ArgumentCaptor.forClass(classOf[Array[Byte]])
+
+    new Lambda().handleRequest(inputStream, outputStream(byteArrayCaptor), null)
+
+    decode[LambdaResponse](byteArrayCaptor.getValue.map(_.toChar).mkString).toOption.get.statusCode should equal(401)
+  }
+
+  "the lambda" should "return 401 if the token has no user id" in {
+    val accessToken = createAccessToken(None)
+    val inputStream = getInputStream(accessToken)
+    val byteArrayCaptor: ArgumentCaptor[Array[Byte]] = ArgumentCaptor.forClass(classOf[Array[Byte]])
+
+    new Lambda().handleRequest(inputStream, outputStream(byteArrayCaptor), null)
+
+    decode[LambdaResponse](byteArrayCaptor.getValue.map(_.toChar).mkString).toOption.get.statusCode should equal(401)
+  }
+
+  "the lambda" should "return 200 if the token is valid" in {
+    val accessToken = createAccessToken(UUID.randomUUID().some)
+    val inputStream = getInputStream(accessToken)
+    val byteArrayCaptor: ArgumentCaptor[Array[Byte]] = ArgumentCaptor.forClass(classOf[Array[Byte]])
+
+    new Lambda().handleRequest(inputStream, outputStream(byteArrayCaptor), null)
+
+    decode[LambdaResponse](byteArrayCaptor.getValue.map(_.toChar).mkString).toOption.get.statusCode should equal(200)
+  }
+
+  "the lambda" should "return allowed origin localhost if the origin is localhost" in {
+    val accessToken = createAccessToken(UUID.randomUUID().some)
+    val inputStream = getInputStream(accessToken)
+    val byteArrayCaptor: ArgumentCaptor[Array[Byte]] = ArgumentCaptor.forClass(classOf[Array[Byte]])
+
+    new Lambda().handleRequest(inputStream, outputStream(byteArrayCaptor), null)
+
+    decode[LambdaResponse](byteArrayCaptor.getValue.map(_.toChar).mkString).toOption.get.statusCode should equal(200)
+  }
+
+  "the lambda" should "return allowed origin integration if the origin is integration" in {
+    val accessToken = createAccessToken(UUID.randomUUID().some)
+    val inputStream = getInputStream(accessToken, "https://tdr-integration.nationalarchives.gov.uk")
+    val byteArrayCaptor: ArgumentCaptor[Array[Byte]] = ArgumentCaptor.forClass(classOf[Array[Byte]])
+
+    new Lambda().handleRequest(inputStream, outputStream(byteArrayCaptor), null)
+
+    decode[LambdaResponse](byteArrayCaptor.getValue.map(_.toChar).mkString).toOption.get.statusCode should equal(200)
+  }
+
+  "the lambda" should "return allowed origin integration if the origin is an unknown domain" in {
```

Comment:
I think this is right because it stops requests from other origins accessing the url. I tried it by changing my local front end to run on port 9001 instead of 9000 and it won't connect to the /cookies endpoint any more, it gives a CORS error.

It's not the greatest security thing ever because you can still use localhost:9000 but it's better than nothing I guess.

Reply:
Yeah, what Sam's described is the correct behaviour. Modern browsers will block JavaScript access to HTTP responses where the allowed origin doesn't match the current site's domain name.

Should there be an assertion about the allowed hosts header in each of these tests to actually check that behaviour? They currently just check the status code.

Reply:
I put it in another comment but I've refactored this a bit so the method that returns the allow origin header is tested separately so we can test all combinations now.